### PR TITLE
Add: Skill calculator items Fix: Prayer bonuses

### PR DESCRIPTION
**Pull request by forsco**
Fixes:  #2917 and #2914

notes marker

notes marker

notes marker

### CODE REVIEW

**Comment by forsco** [comment]
Also adds missing fish.

**Comment by Jalopyy** [comment]
The fishing commit looks like it's missing minnows.

File: runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/beans/SkillDataMultiplier.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package net.runelite.client.plugins.skillcalculator.beans;
+
+public class SkillDataMultiplier
```
[the remark this answers is not in the document]
Comment (forsco) [reply runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/beans/SkillDataMultiplier.java]: Can you please elaborate on this change as I don't know what I'm supposed to be doing.

SkillDataBonus.java, SkillDataEntry follow the exact same convention and weren't created by me.

**Comment by forsco** [comment]
@Jalopyy Minnows have now been added in this PR.

**Comment by forsco** [comment]
6903019

Closes: #2995

**Comment by forsco** [comment]
addBonus is now ignoreBonus @deathbeam

File: runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java [hunk 2]
```diff
@@ -155,19 +156,39 @@ private void clearCombinedSlots()
...
-				uiCheckbox.addActionListener(e -> adjustXPBonus(uiCheckbox.isSelected(), bonus.value));
+				uiCheckbox.addActionListener(e -> adjustXPBonus(uiCheckbox.isSelected(), bonus.getValue()));
+
+				uiOption.add(uiLabel, BorderLayout.WEST);
+				uiOption.add(uiCheckbox, BorderLayout.EAST);
+				add(uiOption);
+			}
+		}
+
+		if (skillData.getMultipliers() != null)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]: Hmm actually, why is this completely different list? i would rather use delimiting boolean probably.

Comment (forsco) [reply runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]:
```java
private void adjustXPBonus(boolean addBonus, float value)
	{
		xpFactor += addBonus ? value : -value;
		calculate();
	}

	private void adjustXPMultiplier (boolean addMultiplier, float value)
	{
		xpFactor = addMultiplier ? value : 1.0f;
		calculate();
	}
```

Comment (forsco) [reply runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]: @deathbeam 
Skilling outfits add a percentage to base xp
Gilded Altar and Ecto are a percentage of base xp

This needed to be seperate in the lists and idendtifiable.

Comment (forsco) [reply runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]: adjustXpBonus is also the original code.

adjustXpMultiplier is the addition so it only affects the prayer calculator.

**Comment by forsco** [comment]
Changes made to remove duplicate renamed methods. Now utilises the original bonuses with an added boolean in the skill JSON to check whether it is a % of base xp, or, base xp + %.

**Comment by forsco** [comment]
Will fix these conflicts in approx 6 hours

**Comment by forsco** [comment]
I will fix these conflicts in appx 6 hrs.

**Comment by forsco** [comment]
Replaced by #3141

**Pull request closed without merging**